### PR TITLE
w262783953 Vice Backend Production Environments

### PR DESCRIPTION
**Pull request by whatisdot**
Closes https://www.wrike.com/open.htm?id=262783953

This enables the use of `--region <region_string>` when running `*_eb` commands and `*_terraform*` commands.

In addition, a [large amount of refactoring](https://github.com/bodyshopbidsdotcom/infra/pull/24) was needed in the infra repository. I updated the `git_terraform_commands` to reflect this.

### Testing

1. Change to your git-scripts directory.
1. Check out this branch.
1. Move to the vice-backend project directory.
1. Try the following commands:

```bash
$ git_terraform_commands --region eu-west-1 zurich

$ git_run_eb --region eu-west-1 zurich rake db:migrate

$ git_setenv_eb --region eu-west-1 zurich

$ git_push_eb --region eu-west-1 zurich
```

Ensure that default/existing environments also work:
```
# ...
$ git_run_eb qa1 rake db:migrate
# ...
```

### CODE REVIEW

File: lib/terraform_commands.py [hunk 1]
```diff
@@ -62,30 +87,62 @@ def env_list(self):
...
+      PROFILE=DEFAULT_PROFILE
     )
 
+  def database_command(self, env_type):
+    ssl_cert = "mysql-ssl-ca-cert.pem"
+    if not self.args.certificate is None:
+      ssl_cert = self.args.certificate
+    elif os.path.isdir('config'):
+      ssl_cert = next(name for name in filter(re.compile('\.pem$').search, os.listdir('config')))
+    ssl_cert = "config/"+ssl_cert
```
Comment (jeet4320) [inline lib/terraform_commands.py]: if there is no certificate path provided and there is not certificate in config folder then we should not add `config/` to the `ssl_cert `

Comment (jeet4320) [reply lib/terraform_commands.py]: Just I saw on `infra` we need `database_ssl_ca_cert ` as argument and it is not default, so it needs some value. Maybe we should deault value to `database_ssl_ca_cert ` in infra

Comment (whatisdot) [reply lib/terraform_commands.py]: It is given a default in `git_terraform_scripts`. I think we should only be using this command to generate our terraform command strings.

**Comment by whatisdot** [comment]
The database_url is output by the configs/database/(production/qa) scripts upon completion:
```bash
module.aws_rds_mysql.aws_db_instance.aws_db_instance: Still creating... (8m50s elapsed)
module.aws_rds_mysql.aws_db_instance.aws_db_instance: Creation complete after 8m51s (ID: vice-backend-qa2)

Apply complete! Resources: 4 added, 0 changed, 0 destroyed.

Outputs:

database_url = mysql2://snapsheet:(redacted)@vice-backend-qa2.ctueul8s6w0k.us-west-2.rds.amazonaws.com:3306/vicebackend?sslca=config/amazon-rds-ca-cert.pem
```

**Comment by whatisdot** [comment]
@abhishek85 Requesting review, because this must enable seniors and leads to deploy to production. Contact me when you have time to test and we can run through it together.

**Pull request merged**